File: snippets/python/unsubscribe_a_contact_from_your_audience.py
import hashlib
import mailchimp_marketing as MailchimpMarketing
from mailchimp_marketing.api_client import ApiClientError

mailchimp = MailchimpMarketing.Client()
mailchimp.set_config({
  "api_key": "YOUR_API_KEY",
  "server": "YOUR_SERVER_PREFIX"
})

list_id = "YOUR_LIST_ID"
member_email = "MEMBER_EMAIL_ADDRESS"
member_email_hash = hashlib.md5(email.encode('utf-8')).hexdigest()
member_update = {"status": "unsubscribed"}

try:
  response = mailchimp.lists.update_list_member(list_id, member_email_hash, member_update)
  print("Response: {}".format(response))
except ApiClientError as error:
  print("An exception occurred: {}".format(error.text))

File: snippets/php/unsubscribe_a_contact_from_your_audience.php
require_once('/path/to/MailchimpMarketing/vendor/autoload.php');

$mailchimp = new \MailchimpMarketing\ApiClient();

$mailchimp->setConfig([
    'apiKey' => 'YOUR_API_KEY',
    'server' => 'YOUR_SERVER_PREFIX'
]);

$email = "prudence.mcvankab@example.com";
$listId = "YOUR_LIST_ID";
$subscriberHash = md5(strtolower($email));

try {
    $response = $mailchimp->lists->updateListMember($listId, $subscriberHash, ["status" => "unsubscribed"]);
    print_r($response);
} catch (MailchimpMarketing\ApiException $e) {
    echo $e->getMessage();
}